src/metabase/query_processor/middleware/permissions.clj

```diff
@@ -90,8 +90,9 @@
         card-id
         (do
           (check-card-read-perms database-id card-id)
-          (when-not (query-perms/check-data-perms outer-query required-perms :throw-exceptions? false)
-            (check-block-permissions outer-query)))
+
+          (when-not (query-perms/has-perm-for-query? outer-query :perms/view-data required-perms)
+            (throw (query-perms/perms-exception required-perms))))
 
         ;; set when querying for field values of dashboard filters, which only require
         ;; collection perms for the dashboard and not ad-hoc query perms
```

Comment:
Can't leave the comment on the right line here, but isn't the below check under `*param-values-query*` doing the same thing as you're changing on L93-95? Should we change it to match?

Reply:
Looks like it. I've updated it to match the more straightforward style.